modules/openapi-generator/src/main/resources/Javascript/es6/ApiClient.mustache

```diff
@@ -404,6 +409,25 @@ class ApiClient {
 
         request.query(this.normalizeParams(queryParams));
 
+    {{#detectAmznApiGatewayExtensions}}
+        // Retrieve host from baseURL and full path from baseURL + path
+        var host = this.basePath.substring(8);
+        var fullPath = host.substring(host.indexOf('/') + 1);
+        host = host.substring( 0, host.indexOf( fullPath ) - 1);
+        fullPath += path;
+
```

Comment:
Can you please look at this to see if you can use that?

https://www.w3schools.com/nodejs/nodejs_url.asp


```
var url = require('url');
var adr = 'http://localhost:8080/default.htm?year=2017&month=february';
var q = url.parse(adr, true);

console.log(q.host); //returns 'localhost:8080'
console.log(q.pathname); //returns '/default.htm'
console.log(q.search); //returns '?year=2017&month=february'

var qdata = q.query; //returns an object: { year: 2017, month: 'february' }
console.log(qdata.month); //returns 'february'
```

Reply:
Looks good, just replaced with this.